packages/number-field/test/validation.test.js

```diff
@@ -0,0 +1,313 @@
+import { expect } from '@esm-bundle/chai';
+import { fixtureSync } from '@vaadin/testing-helpers';
+import sinon from 'sinon';
+import '../src/vaadin-number-field.js';
+
+describe('validation', () => {
+  let field, input;
+
+  describe('basic', () => {
+    beforeEach(() => {
+      field = fixtureSync('<vaadin-number-field></vaadin-number-field>');
+      input = field.inputElement;
+    });
+
+    it('should pass validation when the field by default', () => {
+      expect(field.checkValidity()).to.be.true;
+      expect(field.validate()).to.be.true;
+      expect(field.invalid).to.be.false;
+    });
+
+    it('should not pass validation when the field is required and has no value', () => {
+      field.required = true;
+      expect(field.checkValidity()).to.be.false;
+      expect(field.validate()).to.be.false;
+      expect(field.invalid).to.be.true;
+    });
+
+    it('should pass validation when the field is required and has a valid value', () => {
+      field.required = true;
+      field.value = '1';
+      expect(field.checkValidity()).to.be.true;
+      expect(field.validate()).to.be.true;
+      expect(field.invalid).to.be.false;
+    });
+
+    it('should be valid with numeric values', () => {
+      expect(field.validate()).to.be.true;
```

Comment:
This check can be removed because it's now covered by the newly added test (valid by default).

Reply:
Good catch, removed.